roles/edpm_bootstrap/tasks/swap.yml

```diff
@@ -0,0 +1,59 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Configure swap file
+  become: true
+  when:
+    - not edpm_bootstrap_swap_partition_enabled|bool
+    - edpm_bootstrap_swap_size_megabytes|int > 0
+  shell: |
+    #!/bin/bash
+    set -eu
+    if [ ! -f {{ edpm_bootstrap_swap_path }} ]; then
+      dd if=/dev/zero of={{ edpm_bootstrap_swap_path }} count={{ edpm_bootstrap_swap_size_megabytes }} bs=1M
+      chmod 0600 {{ edpm_bootstrap_swap_path }}
+      mkswap {{ edpm_bootstrap_swap_path }}
+      swapon {{ edpm_bootstrap_swap_path }}
+      if ! grep -qE "{{ edpm_bootstrap_swap_path }}\b" /etc/fstab; then
+        echo "{{ edpm_bootstrap_swap_path }} swap swap defaults 0 0" >> /etc/fstab
+      fi
```

Comment:
I see this has already been discussed, but it would be nice if we can follow up after this is merged to reimplement using: https://docs.ansible.com/ansible/latest/collections/ansible/posix/mount_module.html

Reply:
ack, let's keep this for follow-ups